plugins/inputs/kapacitor/kapacitor.go

```diff
@@ -0,0 +1,228 @@
+	ClusterID        string             `json:"cluster_id"`
+	Host             string             `json:"host"`
+	Kapacitor        *map[string]object `json:"kapacitor"`
+	MemStats         *memstats          `json:"memstats"`
+	NumEnabledTasks  int                `json:"num_enabled_tasks"`
+	NumSubscriptions int                `json:"num_subscriptions"`
+	NumTasks         int                `json:"num_tasks"`
+	Product          string             `json:"product"`
+	ServerID         string             `json:"server_id"`
+	Version          string             `json:"version"`
+}
+
+// Gathers data from a particular URL
+// Parameters:
+//     acc    : The telegraf Accumulator to use
+//     url    : endpoint to send request to
+//
+// Returns:
+//     error: Any error that may have occurred
+func (k *Kapacitor) gatherURL(
+	acc telegraf.Accumulator,
+	url string,
+) error {
+	now := time.Now()
+
+	resp, err := k.client.Get(url)
+	if err != nil {
+		return err
+	}
+	defer resp.Body.Close()
+
+	dec := json.NewDecoder(resp.Body)
+
+	var s stats
+	err = dec.Decode(&s)
+	if err != nil {
+		return err
+	}
+
+	acc.AddFields("kapacitor_memstats",
+		map[string]interface{}{
+			"alloc_bytes":         s.MemStats.Alloc,
+			"buck_hash_sys_bytes": s.MemStats.BuckHashSys,
+			"frees":               s.MemStats.Frees,
+			"gcc_pu_fraction":     s.MemStats.GCCPUFraction,
+			"gc_sys_bytes":        s.MemStats.GCSys,
+			"heap_alloc_bytes":    s.MemStats.HeapAlloc,
+			"heap_idle_bytes":     s.MemStats.HeapIdle,
+			"heap_in_use_bytes":   s.MemStats.HeapInuse,
+			"heap_objects":        s.MemStats.HeapObjects,
+			"heap_released_bytes": s.MemStats.HeapReleased,
+			"heap_sys_bytes":      s.MemStats.HeapSys,
+			"last_gc_ns":          s.MemStats.LastGC,
+			"lookups":             s.MemStats.Lookups,
+			"mallocs":             s.MemStats.Mallocs,
+			"mcache_in_use_bytes": s.MemStats.MCacheInuse,
+			"mcache_sys_bytes":    s.MemStats.MCacheSys,
+			"mspan_in_use_bytes":  s.MemStats.MSpanInuse,
+			"mspan_sys_bytes":     s.MemStats.MSpanSys,
+			"next_gc_ns":          s.MemStats.NextGC,
+			"num_gc":              s.MemStats.NumGC,
+			"other_sys_bytes":     s.MemStats.OtherSys,
+			"pause_total_ns":      s.MemStats.PauseTotalNs,
+			"stack_in_use_bytes":  s.MemStats.StackInuse,
+			"stack_sys_bytes":     s.MemStats.StackSys,
+			"sys_bytes":           s.MemStats.Sys,
+			"total_alloc_bytes":   s.MemStats.TotalAlloc,
+		},
+		map[string]string{
+			"kap_version": s.Version,
+			"url":         url,
+		},
+		now)
+
+	acc.AddFields("kapacitor",
+		map[string]interface{}{
+			"num_enabled_tasks": s.NumEnabledTasks,
+			"num_subscriptions": s.NumSubscriptions,
+			"num_tasks":         s.NumTasks,
+		},
+		map[string]string{
+			"kap_version": s.Version,
+			"url":         url,
+		},
+		now)
+
+	for _, obj := range *s.Kapacitor {
+
+		// Strip out high-cardinality or duplicative tags
+		excludeTags := []string{"host", "cluster_id", "server_id"}
```

Comment:
This struct me as odd, may this behavior can be triggered via config instead of assuming that this information is unnecessary.